src/ssl/gadgets.cc

```diff
@@ -479,18 +479,24 @@ static bool buildCertificate(Security::CertPointer & cert, Ssl::CertificatePrope
         (void)replaceCommonName(cert, properties.commonName);
     }
 
-    // We should get caCert notBefore and notAfter fields and do not allow
-    // notBefore/notAfter values from certToMimic before/after notBefore/notAfter
-    // fields from caCert.
-    // Currently there is not any way in openssl tollkit to compare two ASN1_TIME
-    // objects.
+    ASN1_TIME *notBefore = NULL;
+    ASN1_TIME *notAfter = NULL;
+
+    // Get times from cacert
+    notBefore = X509_getm_notBefore(properties.signWithX509.get());
```

Comment:
The modern C++ way of doing this is

```suggestion
    const auto notBefore = X509_getm_notBefore(properties.signWithX509.get());
```

... but I suggest renaming this variable to clarify that it refers to CA/signing certificate rather than the certificate being mimicked:


```suggestion
    const auto caNotBefore = X509_getm_notBefore(properties.signWithX509.get());
```

Besides other advantages, this version avoids `NULL` which is banned from new/changed Squid code.

However, based on the code further below, properties.signWithX509 may be nil. I would not test how OpenSSL deals with nil pointers and check for that explicitly in Squid code. That probably pushes us one step back:

```suggestion
    ASN1_TIME *notBefore = nullptr;
    if (const auto caCert = properties.signWithX509.get())
        caNotBefore = X509_getm_notBefore(caCert);
```